tests/ludwig/datasets/mnist/test_mnist_workflow.py

```diff
@@ -0,0 +1,72 @@
+import os
+import unittest
+from unittest import mock
+from unittest.mock import mock_open
+import shutil
+from ludwig.datasets.mnist import Mnist
+
+"""TODO: This serves as a placeholder integration test, will be changed into a true unit test soon
+with mocking and patching, commented out for now"""
+
+
+class TestMNistWorkflow(unittest.TestCase):
+
+    def setUp(self):
+        self._cache_dir = "/home/saikatkanjilal/.ludwig_cache"
+        self.dataset = Mnist(cache_dir=self._cache_dir)
+        self.raw_dataset_path = os.path.join(self._cache_dir, "mnist_1.0", "raw")
+        self.raw_tmp_path = os.path.join(self._cache_dir, "mnist_1.0", "raw")
+        self.processed_dataset_path = os.path.join(self._cache_dir, "mnist_1.0", "processed")
+        self.cleanup(self.raw_tmp_path)
+        self.cleanup(self.raw_dataset_path)
+        self.cleanup(self.processed_dataset_path)
+
+    def cleanup(self, delete_path):
+        for root, dirs, files in os.walk(delete_path):
+            for name in files:
+                # make sure what you want to keep isn't in the full filename
+                os.remove(os.path.join(delete_path , name))
+            for name in dirs:
+                shutil.rmtree(os.path.join(delete_path, name))
+
+    def tearDown(self):
+        """self.cleanup(self.raw_tmp_path)
+        self.cleanup(self.raw_dataset_path)
+        self.cleanup(self.processed_dataset_path)"""
+        assert True
+
+    @mock.patch("builtins.open", create=True)
+    @mock.patch("gzip.GzipFile.read")
+    @mock.patch("gzip.GzipFile")
+    @mock.patch("requests.get")
+    def test_mnist_download(self,
```

Comment:
Similar to the other tests, can we generate some synthetic data to test this out?

Reply:
See latest test, based on our discusssion lets figure out what more we can do to solidify this